src/bomf/filter/__init__.py

```diff
@@ -34,20 +34,23 @@ async def predicate(self, candidate: Candidate) -> bool:
         """
         raise NotImplementedError("The inheriting class has to implement this method")
 
-    async def apply(self, candidates: List[Candidate]) -> List[Candidate]:
+    async def apply(self, candidates: Iterable[Candidate]) -> Iterable[Candidate]:
         """
         apply this filter on the candidates
         """
         tasks: List[Awaitable[bool]] = [self.predicate(c) for c in candidates]
         self._logger.info("%s created %i predicate tasks; Awaiting them all", str(self), len(tasks))
         predicate_results = await asyncio.gather(*tasks)
         self._logger.info("%s awaited %i tasks", str(self), len(tasks))
-        result = [
+        result = (
             c for c, predicate_match in zip(candidates, predicate_results, strict=True) if predicate_match is True
-        ]
+        )
```

Comment:
Kannst du candidates hier in den zip einfach so reinpacken? Du iterierst ja schon in Zeile 41 da durch, wenn das jetzt wirklich ne generator function ist, wird das doch nicht funktionieren, oder?